TestTicTacToe/GameRulesTest.cs

```diff
@@ -0,0 +1,219 @@
+using TicTacToe;
+using Xunit;
+
+namespace TestTicTacToe
+{
+    public class GameRulesTest
+    {
+
+        [Fact]
+        public void Should_Return_False_When_3_In_A_Row_Horizontally_Of_Different_Players_Marks()
+        {
+            // Arrange
+            var gameLosingBoard = new[,]
+            {
+                {1, 2, 1},
+                {0, 0, 0},
+                {0, 0, 0}
+            };
+        
+            // Act
+            var hasWonGame = GameRulesHandler.HasWon(gameLosingBoard);
+        
+            // Assert
+            Assert.False(hasWonGame);
+        }
+        
+        
+        [Fact]
+        public void Should_Return_False_When_3_In_A_Row_Vertically_Of_Different_Players_Marks()
+        {
+            // Arrange 
+            var gameLosingBoard = new[,]
+            {
+                {0, 1, 0},
+                {0, 1, 0},
+                {0, 2, 0}
+            };       
+        
+            // Act
+            var hasWonGame = GameRulesHandler.HasWon(gameLosingBoard);
+        
+            // Assert
+            Assert.False(hasWonGame);
+        }
+
+        [Fact]
+        public void Should_Return_False_When_3_In_A_Row_Diagonally_As_Forward_Slash_Of_Different_Players_Marks()
+        {
+            // Arrange 
+            var gameLosingBoard = new[,]
+            {
+                {2, 0, 0},
+                {0, 1, 0},
+                {0, 0, 1}
+            };
+        
+        
+            // Act
+            var hasWonGame = GameRulesHandler.HasWon(gameLosingBoard);
+        
+            // Assert
+            Assert.False(hasWonGame);
+        }
+        
+        [Fact]
+        public void Should_Return_False_When_3_In_A_Row_Diagonally_As_Backward_Slash_Of_Different_Players_Marks()
+        {
+            // Arrange 
+            var gameLosingBoard = new[,]
+            {
+                {0, 0, 1},
+                {0, 2, 0},
+                {1, 0, 0}
+            };
+        
+        
+            // Act
+            var hasWonGame = GameRulesHandler.HasWon(gameLosingBoard);
+        
+            // Assert
+            Assert.False(hasWonGame);
+        }
+        
+        
+        [Fact]
+        public void Should_Return_True_When_GameStatus_Is_Checked_If_3_In_A_Row_Horizontally()
```

Comment:
I think you need to add in the fact that they are 3 of the same token in row, otherwise it sounds that you win once three of any tokens are placed in a row